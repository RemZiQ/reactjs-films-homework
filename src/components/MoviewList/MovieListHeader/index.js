export { default } from './MovieListHeader.container';
