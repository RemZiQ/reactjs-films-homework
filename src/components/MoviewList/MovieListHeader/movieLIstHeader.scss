.movieListNav {
  padding: 60px 100px;
  background-color: #f1f1f1;
}

.movieListNav ul {
  border-bottom: 1px solid rgb(173, 173, 173);
}

.movieListNav li {
  display: inline-block;
}

.movieListNav button {
  margin-right: 40px;
  border-style: none;
  background-color: rgba(0, 0, 0, 0);
  border-bottom: 2px solid rgba(0, 0, 0, 0);
  font-size: 28px;
  letter-spacing: 1px;
  line-height: 44px;
  color: #0aaee4;
  font-family: "Helvetica Neue";
  font-weight: bold;
}

.movieListNav button:hover {
  border-bottom: 2px solid #0aaee4;
}

.movieListNav button:focus {
  outline: none;
  border-bottom: 2px solid #0aaee4;
}

.dropDownGenres a {
  text-decoration: none;
}
.dropDownGenres li {
  display: list-item;
  padding: 5px 20px;
  font-size: 28px;
  letter-spacing: 1px;
  line-height: 44px;
  color: #0aaee4;
  font-family: "Helvetica Neue";
  font-weight: 500;
}

.dropDownGenres li:hover {
  background-color: rgba(10, 174, 228, 0.25);
}
.dropDownGenres{
  display: none;
  position: absolute;
  top: 100%;
  background-color: #f1f1f1;
  z-index: 100;
}

.dropDown {
  position: relative;
}
.dropDown:hover > ul {
  display: block;
}

// .movieListNav select {
//   display: inline-block;
//   background-color: #f1f1f1;
//   color: #0aaee4;
//   outline: none;
//   border-style: none;
//   font-size: 28px;
//   letter-spacing: 1px;
//   line-height: 44px;
//   font-family: "Helvetica Neue";
//   font-weight: bold;
// }
// .movieListNav option {
//   display: block;
// }

// .movieListNav select {
//   -webkit-appearance: none;
//   -moz-appearance: none;
//   appearance: none;
// }