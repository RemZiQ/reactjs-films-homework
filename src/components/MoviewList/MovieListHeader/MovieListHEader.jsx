import React from 'react';
import { Link } from 'react-router-dom';
import propTypes from 'prop-types';


import './movieLIstHeader.scss';

const MovieListHeader = ({
  getTrandingFilms,
  getTopRated,
  getComingSoonFilms,
  getByGenre,
  genres,
}) => {
  const handleForFetchByGenre = (e) => {
    getByGenre(e.target.id);
  };
  const listOfButtonsByGenres = genres.map(item => (
    <Link to={{ pathname: '/search', search: `?query=${item.name}` }}>
      <li onClick={handleForFetchByGenre} id={item.id}>{item.name}</li>
    </Link>
  ));
  return (
    <nav className="movieListNav">
      <ul>
        <li>
          <Link to={{ pathname: '/search', search: '?query=Trending' }}>
            <button onClick={getTrandingFilms} type="submit">Trending</button>
          </Link>
        </li>
        <li>
          <Link to={{ pathname: '/search', search: '?query=Top Rated' }}>
            <button onClick={getTopRated} type="submit">Top Rated</button>
          </Link>
        </li>
        <li>
          <Link to={{ pathname: '/search', search: '?query=Coming soon' }}>
            <button onClick={getComingSoonFilms} type="submit">Coming soon</button>
          </Link>
        </li>
        <li className="dropDown">
          <button type="submit">Genre</button>
          <ul className="dropDownGenres">
            {listOfButtonsByGenres}
          </ul>
        </li>
      </ul>
    </nav>
  );
};


MovieListHeader.propTypes = {
  getTrandingFilms: propTypes.func.isRequired,
  getTopRated: propTypes.func.isRequired,
  getComingSoonFilms: propTypes.func.isRequired,
  getByGenre: propTypes.func.isRequired,
  genres: propTypes.arrayOf(propTypes.object),
};

MovieListHeader.defaultProps = {
  genres: [{ id: 28, name: 'Action' }],
};

// MovieListHeader.defaultProps = {
//   fetchTrandingData: null,
//   fetchTopRatedData: null,
//   fetchComingSoonData: null,
// };

export default MovieListHeader;
