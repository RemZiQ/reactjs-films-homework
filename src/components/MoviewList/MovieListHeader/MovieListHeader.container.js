import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import {
  getTrandingFilms,
  getTopRated,
  getComingSoonFilms,
  getByGenre,
} from '../../../modules/module_Search/actions';
import { genres } from '../../../modules/module_Search/selectors';
import MovieListHeader from './MovieListHEader';

const mapStateToProps = createStructuredSelector({ genres });

const mapDispatchToProps = {
  getTrandingFilms,
  getTopRated,
  getComingSoonFilms,
  getByGenre,
};


export default connect(mapStateToProps, mapDispatchToProps)(MovieListHeader);
