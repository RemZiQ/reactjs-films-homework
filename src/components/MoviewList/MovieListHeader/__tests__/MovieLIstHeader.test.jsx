import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import MovieListHeader from '../MovieListHEader';


it('MovieListHeader renders well without props', () => {
  const renderer = new ShallowRenderer();
  const result = renderer.render(<MovieListHeader />);
  expect(result).toMatchSnapshot();
});

it('MovieListHeader renders well with props', () => {
  const props = {
    genres: [
      {
        name: 'Action',
        id: 22,
      },
      {
        name: 'Action2',
        id: 23,
      },
      {
        name: 'Action3',
        id: 24,
      },
    ],
  };
  const renderer = new ShallowRenderer();
  const result = renderer.render(<MovieListHeader {...props} />);
  expect(result).toMatchSnapshot();
});
