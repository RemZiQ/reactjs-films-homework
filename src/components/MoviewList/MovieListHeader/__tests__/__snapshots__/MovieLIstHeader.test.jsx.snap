// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieListHeader renders well with props 1`] = `
<nav
  className="movieListNav"
>
  <ul>
    <li>
      <Link
        to={
          Object {
            "pathname": "/search",
            "search": "?query=Trending",
          }
        }
      >
        <button
          type="submit"
        >
          Trending
        </button>
      </Link>
    </li>
    <li>
      <Link
        to={
          Object {
            "pathname": "/search",
            "search": "?query=Top Rated",
          }
        }
      >
        <button
          type="submit"
        >
          Top Rated
        </button>
      </Link>
    </li>
    <li>
      <Link
        to={
          Object {
            "pathname": "/search",
            "search": "?query=Coming soon",
          }
        }
      >
        <button
          type="submit"
        >
          Coming soon
        </button>
      </Link>
    </li>
    <li
      className="dropDown"
    >
      <button
        type="submit"
      >
        Genre
      </button>
      <ul
        className="dropDownGenres"
      >
        <Link
          to={
            Object {
              "pathname": "/search",
              "search": "?query=Action",
            }
          }
        >
          <li
            id={22}
            onClick={[Function]}
          >
            Action
          </li>
        </Link>
        <Link
          to={
            Object {
              "pathname": "/search",
              "search": "?query=Action2",
            }
          }
        >
          <li
            id={23}
            onClick={[Function]}
          >
            Action2
          </li>
        </Link>
        <Link
          to={
            Object {
              "pathname": "/search",
              "search": "?query=Action3",
            }
          }
        >
          <li
            id={24}
            onClick={[Function]}
          >
            Action3
          </li>
        </Link>
      </ul>
    </li>
  </ul>
</nav>
`;

exports[`MovieListHeader renders well without props 1`] = `
<nav
  className="movieListNav"
>
  <ul>
    <li>
      <Link
        to={
          Object {
            "pathname": "/search",
            "search": "?query=Trending",
          }
        }
      >
        <button
          type="submit"
        >
          Trending
        </button>
      </Link>
    </li>
    <li>
      <Link
        to={
          Object {
            "pathname": "/search",
            "search": "?query=Top Rated",
          }
        }
      >
        <button
          type="submit"
        >
          Top Rated
        </button>
      </Link>
    </li>
    <li>
      <Link
        to={
          Object {
            "pathname": "/search",
            "search": "?query=Coming soon",
          }
        }
      >
        <button
          type="submit"
        >
          Coming soon
        </button>
      </Link>
    </li>
    <li
      className="dropDown"
    >
      <button
        type="submit"
      >
        Genre
      </button>
      <ul
        className="dropDownGenres"
      >
        <Link
          to={
            Object {
              "pathname": "/search",
              "search": "?query=Action",
            }
          }
        >
          <li
            id={28}
            onClick={[Function]}
          >
            Action
          </li>
        </Link>
      </ul>
    </li>
  </ul>
</nav>
`;
